monitor/src/main.rs

```diff
@@ -8,10 +8,10 @@ use crate::controller::send_to_controller;
 use clap::{Arg, Command};
 use std::path::Path;
 use std::sync::Arc;
-use tokio::runtime::Runtime;
+use tokio;
 
-
-fn main() {
+#[tokio::main]
+async fn main() {
```

Comment:
Isn't it dangerous to make the main function asynchronous?

Reply:
Making the main function asynchronous is actually a common practice in Rust applications that use asynchronous runtimes like tokio. It allows you to leverage asynchronous programming and concurrency in a straightforward manner. Whn using tokio, the main function is almost always async